### ISSUE TITLE
Extend HTTP API to retrieve filtered tiddlers

### ISSUE THREAD

**Issue by Jermolene** [Jermolene/TiddlyWiki5#4054]
As [discussed here](https://groups.google.com/d/msgid/tiddlywiki/9cf91e41-dc22-4041-901d-b93bf059c64a%40googlegroups.com?utm_medium=email&utm_source=footer) it would be useful to be able to query the TW5 server for a filtered list of tiddlers.

For example, we could extend the `/recipes/default/tiddlers.json` endpoint with an optional `filter` parameter. For example, to retrieve the tiddlers resulting from the filter `[tag[mytag]]`:

```
GET /recipes/default/tiddlers.json?filter=%5Btag%5Bmytag%5D%5D
```

**Comment by inmysocks** [comment]
I have a setup that is not completely unlike this as part of the Bob server. I am not sure how much of it can be directly copied into the tiddlywiki server but it should be a place to start. I will look at what I have and see if I can do a quick proof of concept with the built-in server.

**Comment by bimlas** [comment]
@Jermolene, I've just opened a PR, please check it out.

**Comment by bimlas** [comment]
Just created a userscript to combine Google and TiddlyWiki search results:

https://github.com/bimlas/userscript-combine-tw5-and-search-engine-results

The webserver API filter option allows you to search for more than one wiki at a time. This could somehow be used in TW-Bob in my opinion (or it is already implemented @inmysocks ?)

**Comment by inmysocks** [comment]
Bob has a way tofetch a filtered list of tiddlers from a wiki. It is a bit more complex because of the multiple wikis and authentication used.

I need to look at the search engine results script you made to see if I can see if it would work with Bob.

I like the idea, it looks like a good step toward federation in the core.

**Comment by Jermolene** [comment]
This is now fixed in #4373
